Source/objects.cpp

```diff
@@ -1345,14 +1342,35 @@ void AddTrap(Object &trap)
 	trap._oVar4 = 0;
 }
 
-void AddObjectLight(Object &object, int r)
+void AddObjectLight(Object &object)
 {
-	if (ApplyObjectLighting) {
-		DoLighting(object.position, r, {});
-		object._oVar1 = -1;
-	} else {
-		object._oVar1 = 0;
+	int radius;
+	switch (object._otype) {
+	case OBJ_STORYCANDLE:
+	case OBJ_L5CANDLE:
+		radius = 3;
+		break;
+	case OBJ_L1LIGHT:
+	case OBJ_SKFIRE:
+	case OBJ_CANDLE1:
+	case OBJ_CANDLE2:
+	case OBJ_BOOKCANDLE:
+	case OBJ_BCROSS:
+	case OBJ_TBCROSS:
+		radius = 5;
+		break;
+	case OBJ_TORCHL:
+	case OBJ_TORCHR:
+	case OBJ_TORCHL2:
+	case OBJ_TORCHR2:
+		radius = 8;
+		break;
+	default:
+		return;
 	}
+
+	DoLighting(object.position, radius, {});

```

Comment:
When porting to lazarus (books are activated) `LoadingMapObjects` is true.
That means `DoLighting` update `dPreLight`.
But `dLight` is not yet updated. The new light source is visible only after another lightning update happens for the relevant tiles (for example hero or monster moving).

![grafik](https://user-images.githubusercontent.com/25415264/233857657-d464b5d2-37f3-48c5-a6e9-b430eba1bbcd.png)

Reply:
fixed :)